tests/testsuite/messages.rs

```diff
@@ -48,6 +46,20 @@ pub fn raw_rustc_output(project: &Project, path: &str, extra: &[&str]) -> String
     result
 }
 
+fn redact_rustc_message(msg: &str) -> String {
+    let mut msg = msg
+        .replace("help:", "[HELP]")
+        .replace("note:", "[NOTE]")
+        .replace("warning:", "[WARNING]")
+        .replace("error:", "[ERROR]");
```

Comment:
I'd rather we provide a way to expose these redactions and run them on `msg`.  For example, we could add a getter to `snapbox::Assert`